Lib/shutil.py

```diff
@@ -605,7 +587,13 @@ def _rmtree_unsafe(path, onerror):
         entries = []
     for entry in entries:
         fullname = entry.path
-        if _rmtree_isdir(entry):
+        try:
+            is_dir = (entry.is_dir(follow_symlinks=False) and
+                      not entry.is_junction())
+        except OSError:
+            is_dir = False
+
+        if is_dir:
```

Comment:
Since the `is_junction()` method call can't raise `OSError`, you could move that check out of the try/except. I think this is less muddled since a junction is a directory.

```suggestion
        try:
            is_dir = entry.is_dir(follow_symlinks=False)
        except OSError:
            is_dir = False

        if is_dir and not entry.is_junction():
```